presentation/frontend/src/presentation-frontend/PresentationManager.ts

```diff
@@ -241,6 +242,10 @@ export class PresentationManager implements IDisposable {
       this._clearEventListener = IpcApp.addListener(PresentationIpcEvents.Update, this.onUpdate);
       this._ipcRequestsHandler = props?.ipcRequestsHandler ?? new IpcRequestsHandler(this._requestsHandler.clientId);
     }
+
+    PresentationManagerExtensions.provider = {
+      getContentObservable: this.getContentObservable.bind(this),
+    };
```

Comment:
In theory you could have multiple `PresentationManager` instances set up with different attributes (active unit system, locale, etc.). In that case, all those instances would use `PresentationManagerExtensions` set up with the most recently created `PresentationManager` attributes. Also, it's not great that developers need to know that `PresentationManager` must be created before `PresentationManagerExtensions` can be used at all.

My suggestion:
1. Change `PresentationManagerExtensions` to accept implementation through constructor.
2. Change `PresentationManagerExtensions.getContentObservable` to be non static.
3. Create `PresentationManagerExtensions` object as a member when creating a `PresentationManager`. Add an internal public getter for it.
4. `HiliteSetProvider` can then use it like `Presentation.presentation.internal.getContentObservable(...)`.
5. I'd also rename `PresentationManagerExtensions` to `PresentationManagerInternal` and `PresentationManagerProvider` to `PresentationManagerInternalImpl`.

Reply:
Made it so that HiliteSetProvider uses an iterator as we've discussed